src/lib/language.ml

```diff
@@ -230,6 +265,39 @@ let rec to_shell: type a. _ -> a t -> string =
       Literal.to_shell lit
     | Output_as_string e ->
       sprintf "\"$( { %s ; } | od -t o1 -An -v | tr -d ' \\n' )\"" (continue e)
+    | Int_to_string i ->
+      continue (Output_as_string (Raw_cmd (sprintf "printf '%%d' %s" (continue i))))
+    | String_to_int s ->
+      let var = "tmpxxxxijljlijdeifh" in
+      let value = sprintf "\"$%s\"" var in
+      sprintf " $( %s=$( %s ) ; if [ %s -eq %s ] ; then printf -- %s ; else %s ; fi ; ) "
+        var
```

Comment:
I do not understand, can you document a little bit what is going on here?
I guess specifically when `-eq` will be false when comparing this variable.